src/games/oware.rs

```diff
@@ -0,0 +1,247 @@
+//! Oware is a strategy game from the family of mancala
+//!
+//! It has no  combos, i.e. one turn each round each player.
+//! Official version contains 12 pits and 4 initial seeds in each pit. This implementation is const generic parameterized with PITS and INIT_SEEDS.
+//!
+//! # Rules
+//! 1. Each player has control over half the pits that are in front of them
+//! 2. Player has to choose a non-empty pit. This removes the seeds from that pit and
+//!    distributes them, dropping one in each pit, except the selected pit,
+//!    in counter-clock wise.
+//! 3. If the last seed is dropped in one of the opponents pit, and the final seed count in that
+//!    pit is 2 or 3, then the current player captures it. It can happen several times if
+//!    continous multiple pits to the last dropped pit also have `2` or `3` seeds in them.
+//!    If all the seeds could be captured this turn, then no capture happens
+//!
+//! Objective is to capture as many seeds as possible and so the game will conclude early if a
+//! player captures more than half the seeds
+use std::fmt::{Debug, Display, Formatter};
+
+use internal_iterator::{Internal, IteratorExt};
+
+use crate::board::{Alternating, Board, BoardMoves, BruteforceMoveIterator, Outcome, Player};
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct OwareBoard<const PITS: usize, const INIT_SEEDS: u8> {
+    pits: [u8; PITS],
+    scores: [u8; 2],
```

Comment:
Maybe add some way to access these values? They're probably necessary to implement interesting bots or even just the `Heuristic` trait.

Something like `fn score(&self, player: Player) -> u8` and similar for `pits`?

Reply:
For pits, I have implemented the `Index` trait where the indices always start with respect to the `next_player`. So if the `next_player` is a or b, `0..PPP` will be of the players, and `PPP..PPP*2` will be that of the other player.

For the score, I will add that function. I haven't yet explored the `Heuristic` part of the crate.

Reply:
Ah could you do something that also takes a player as an argument instead? That's a bit more clear than having to rely on the order of players and hoping you don't accidentally go out of bounds.

Something like `fn pits(&self, player: Player, index: usize)` -> u8`.

Reply:
Added `get_seeds` fn get seeds from a pit that the player owns